### PR TITLE
Added SecretManager filtered-listing samples

### PR DESCRIPTION
**Pull request by gwhitehawk**
Adds code samples for listSecrets and listSecretVersions with filter set.

- [ x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md)
- [ x] `pom.xml` parent set to latest `shared-configuration`
- [ x] Appropriate changes to README are included in PR
- [ x] API's need to be enabled to test (tell us)
- [ ] Environment Variables need to be set (ask us to set them)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [ ] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: secretmanager/src/main/java/secretmanager/ListSecretsWithFilter.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient.ListSecretsPagedResponse;
+import java.io.IOException;
+
+public class ListSecretsWithFilter {
+
+  public static void listSecrets() throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String filter = "name:your-secret-substring AND expire_time<2022-01-01T00:00:00Z";
```
Comment (averikitsch) [inline secretmanager/src/main/java/secretmanager/ListSecretsWithFilter.java:31]: nit: for this and the other filter sample can you link to any docs on how to create these filters?

**Review by averikitsch — APPROVED** [review]

File: secretmanager/src/main/java/secretmanager/ListSecretsWithFilter.java (hunk 1 above)
Comment (gwhitehawk) [reply secretmanager/src/main/java/secretmanager/ListSecretsWithFilter.java:33]: Done

**Comment by lesv** [comment]
@gwhitehawk If you visit `github/`, you can join the `googleapis` & `GoogleCloudPlatform` orgs, you'll then be able to start kokoro automatically.

**Review by lesv — APPROVED** [review]

**Comment by lesv** [comment]
@gwhitehawk I approved since Averi had already approved.  Please fix the lint issue, and ping me if you want me to merge for you.

**Comment by gwhitehawk** [comment]
Could you please apply the label to run kokoro jobs?

**Pull request merged**